packages/xrpl/src/Wallet/index.ts

```diff
@@ -318,13 +320,26 @@ class Wallet {
       multisignAddress = this.classicAddress
     }
 
-    if (transaction.TxnSignature || transaction.Signers) {
+    const tx = { ...transaction }
+
+    if (tx.TxnSignature || tx.Signers) {
       throw new ValidationError(
         'txJSON must not contain "TxnSignature" or "Signers" properties',
       )
     }
 
-    const txToSignAndEncode = { ...transaction }
+    // Remove trailing insignificant zeros for non-XRP amount
+    if (tx.TransactionType === 'Payment' && tx.Amount) {
+      if (
+        typeof tx.Amount !== 'string' &&
+        TRAILING_ZEROS_REGEX.exec(tx.Amount.value) !== null
+      ) {
+        tx.Amount = { ...tx.Amount }
```

Comment:
Why is this step necessary?

Reply:
But we're setting it equal to itself, so wouldn't it still modify the `transaction` param?

Reply:
If `tx.Amount` is an object, it still refers to `transaction.Amount` since it copied the reference to the object. Therefore, we need to do a deep copy of `tx.Amount` that won't reference `transaction.Amount` object.